tests/data/test_converter_public_trades.py

```diff
@@ -0,0 +1,288 @@
+                                            'max_delta', 'total_trades',
+                                            'stacked_imbalances_bid',
+                                            'stacked_imbalances_ask']
+
+    assert -50.519000000000005 == result['delta']
+    assert 219.961 == result['bid']
+    assert 169.442 == result['ask']
+
+    assert 151 == len(result.trades)
+    t = result['trades'].iloc[0]
+    assert trades['id'][0] == t["id"]
+    assert int(trades['timestamp'][0]) == int(t['timestamp'])
+    assert 'sell' == t['side']
+    assert '313881442' == t['id']
+    assert 234.72 == t['price']
+
+
+def test_public_trades_put_volume_profile_into_ohlcv_candles(public_trades_list_simple, candles):
+    df = public_trades_to_dataframe(public_trades_list_simple,  'doesntmatter')
+    df = trades_to_volumeprofile_with_total_delta_bid_ask(
+        df, scale=BIN_SIZE_SCALE)
+    candles['vp'] = np.nan
+    candles.loc[candles.index == 1, ['vp']] = candles.loc[candles.index == 1, [
+        'vp']].applymap(lambda x: pd.DataFrame(df.to_dict()))
+    assert 0.14 == candles['vp'][1].values.tolist()[1][2]  # delta
+    assert 0.14 == candles['vp'][1]['delta'].iat[1]
+
+
+def test_public_trades_binned_big_sample_list(public_trades_list):
+    BIN_SIZE_SCALE = 0.05
+    trades = public_trades_to_dataframe(public_trades_list,  'doesntmatter')
+    df = trades_to_volumeprofile_with_total_delta_bid_ask(
+        trades, scale=BIN_SIZE_SCALE)
+    assert df.columns.tolist() == ['bid', 'ask', 'delta',
+                                   'bid_amount', 'ask_amount',
+                                   'total_volume', 'total_trades']
+    assert 23 == len(df)
+    assert df.index[0] < df.index[1] < df.index[2]
+    assert df.index[0] + BIN_SIZE_SCALE == df.index[1]
+    assert (trades['price'].min() -
+            BIN_SIZE_SCALE) < df.index[0] < trades['price'].max()
+    assert (df.index[0] + BIN_SIZE_SCALE) >= df.index[1]
+    assert (trades['price'].max() -
+            BIN_SIZE_SCALE) < df.index[-1] < trades['price'].max()
+
+    assert 32 == df['bid'].iat[0]  # bid
+    assert 197.512 == df['bid_amount'].iat[0]  # bid
+    assert 88.98 == df['ask_amount'].iat[0]  # ask
+    assert 26 == df['ask'].iat[0]  # ask
+    assert -108.53200000000001 == df['delta'].iat[0]  # delta
+    assert 3 == df['bid'].iat[-1]  # bid
+    assert 50.659 == df['bid_amount'].iat[-1]  # bid
+    assert 108.21 == df['ask_amount'].iat[-1]  # ask
+    assert 44 == df['ask'].iat[-1]  # ask
+    assert 57.551 == df['delta'].iat[-1]  # delta
+
+    BIN_SIZE_SCALE = 1
+    trades = public_trades_to_dataframe(public_trades_list, 'doesntmatter')
+    df = trades_to_volumeprofile_with_total_delta_bid_ask(
+        trades, scale=BIN_SIZE_SCALE)
+    assert 2 == len(df)
+    assert df.index[0] < df.index[1]
+    assert (trades['price'].min() -
+            BIN_SIZE_SCALE) < df.index[0] < trades['price'].max()
```

Comment:
I think  these tests (in general) would benefit from some commenting ... 
why would i expect the min price - 1 to be smaller than index[0] ... and so on